applications/app/controllers/AllIndexController.scala

```diff
@@ -105,19 +121,32 @@ class AllIndexController(contentApiClient: ContentApiClient, sectionsLookUp: Sec
           date,
           tzOverride = None
         )
-      ).orElse(item.tag.map( apitag => {
-        val tag = Tag.make(apitag)
-        IndexPage(page = tag, contents = item.results.getOrElse(Nil).map(IndexPageItem(_)), Tags(List(tag)), date, tzOverride = None)
-      }))
+      ).orElse {
+        item.tag.map { apitag =>
+          val tag = Tag.make(apitag)
+          IndexPage(
+            page = tag,
+            contents = item.results.getOrElse(Nil).map(IndexPageItem(_)),
+            Tags(List(tag)),
+            date,
+            tzOverride = None
+          )
+        }
+      }
     }
 
-    result.recover{ case e: Exception =>
-      log.error(s"Cannot fetch content for path '$path'", e)
-      None
+    result.onFailure {
+      case GuardianContentApiError(404, _, _) =>
+        log.warn(s"Cannot fetch content for request '${request.uri}'")
+      case e: Exception =>
+        log.error(e.getMessage, e)
     }
+
+    result
```

Comment:
I dissociated logging and error handling.
I also move the error handling to the caller. Let me know how you feel about this